code/procs/mobprocs/chatprocs.dm

```diff
@@ -426,6 +428,42 @@
 	if (src.get_brain_damage() >= 60)
 		speechverb = pick("says","stutters","mumbles","slurs")
 
+	if (src.singing && src.bioHolder && src.bioHolder.HasEffect("accent_scots"))
+		text = danny_boy(src.danny_index)
+	if (src.singing || (src.bioHolder && src.bioHolder.HasEffect("accent_elvis")))
+		// use note icons instead of normal quotes
+		var/note_type = src.singing == "bad" ? "notebad" : "note"
+		var/note_img = "<img class=\"icon misc\" style=\"position: relative; bottom: -3px; \" src=\"[resource("images/radio_icons/[note_type].png")]\">"
+		if (singing == "loud" || singing == "bad" || loudness > 0)
+			first_quote = "[note_img][note_img]"
+			second_quote = first_quote
+		else
+			first_quote = note_img
+			second_quote = note_img
+		// select appropriate singing verb
+		if (src.bioHolder && src.bioHolder.HasEffect("smoker"))
+			speechverb = "rasps"
+			if (singing == "loud" || loudness > 0)
+				speechverb = "sings Tom Waits style"
+		else if (src.traitHolder && src.traitHolder.hasTrait("french"))
+			speechverb = "sings [pick("Charles Trenet", "Serge Gainsborough", "Edith Piaf")] style"
+		else if (src.bioHolder && src.bioHolder.HasEffect("accent_swedish"))
+			speechverb = "sings disco style"
+		else if (src.bioHolder && src.bioHolder.HasEffect("accent_scots"))
+			speechverb = pick("laments", "softly sings", "croons", "sorrowfully intones", "sobs", "bemoans")
+		else if (src.bioHolder && src.bioHolder.HasEffect("accent_chav"))
+			speechverb = "raps grime"
```

Comment:
grammar is off

Reply:
for which verbs?

Reply:
John Smith laments, "Oh Danny Boy..."
John Smith raps grime, "Oh, It's real out here like no one understands sometimes"
seems ok to me?

Reply:
changed "raps grime" to "grimily raps"